### ISSUE TITLE
0 hits for line containing `done <<eof` in a shell script even though it is actually hit

### ISSUE THREAD

**Issue by susam** [SimonKagstrom/kcov#152]
Here is an example shell script that may be used to reproduce the issue.
```
# Test 1: Coverage works fine
printf "foo\nbar\n" | while read -r line
do
    echo line: "$line"
done

# Test 2: Coverage missed for line with here-document
while read -r line
do
    echo line: "$line"
done <<eof
baz
qux
eof
```
Save the above code in a file, say `foo.sh`. Then execute the following command.
```
kcov coverage foo.sh
```
Since every line of the script was executed, the coverage result should show 100% coverage. But the coverage result shows that `done <<eof` was not executed.

Here is the output of the `kcov` command and the coverage result.
```
$ kcov coverage foo.sh
line: foo
line: bar
line: baz
line: qux
$ cat coverage/foo.sh/foo.sh.*.json
var data = {lines:[
{"lineNum":"    1","line":"# Test 1: Coverage works fine"},
{"lineNum":"    2","line":"printf \"foo\\nbar\\n\" | while read -r line","class":"lineCov","hits":"4","order":"1","possible_hits":"0",},
{"lineNum":"    3","line":"do"},
{"lineNum":"    4","line":"    echo line: \"$line\"","class":"lineCov","hits":"2","order":"2","possible_hits":"0",},
{"lineNum":"    5","line":"done"},
{"lineNum":"    6","line":""},
{"lineNum":"    7","line":"# Test 2: Coverage missed for line with here-document"},
{"lineNum":"    8","line":"while read -r line","class":"lineCov","hits":"3","order":"3","possible_hits":"0",},
{"lineNum":"    9","line":"do"},
{"lineNum":"   10","line":"    echo line: \"$line\"","class":"lineCov","hits":"2","order":"4","possible_hits":"0",},
{"lineNum":"   11","line":"done <<eof","class":"lineNoCov","hits":"0","possible_hits":"0",},
{"lineNum":"   12","line":"baz"},
{"lineNum":"   13","line":"qux"},
{"lineNum":"   14","line":"eof"},
]};
var percent_low = 25;var percent_high = 75;
var header = { "command" : "foo.sh", "date" : "2016-09-25 11:39:36", "instrumented" : 5, "covered" : 4,};
var merged_data = [];
```
![foo-coverage](https://cloud.githubusercontent.com/assets/229488/18813322/5911e8b4-8314-11e6-8347-d115fbc50752.png)

It looks like kcov is unable to recognize that the `done <<eof` line should not be used in the code-coverage computation.

**Comment by SimonKagstrom** [comment]
Thanks for the detailed report! Very clear and well-written!

The long-term solution for this is to implement Issue #145, but in the meantime I've added a regression test for it and it should be possible to fix fairly easy even with the current non-parser.

**Comment by SimonKagstrom** [comment]
Resolved with 6ca90c9. Ugly fix though :-)
